bindings/nodejs/lib/types/block/address.ts

```diff
@@ -118,12 +133,89 @@ class NftAddress extends Address {
     }
 }
 
+/**
+ * An implicit account creation address.
+ */
+class ImplicitAccountCreationAddress extends Address {
+    /**
+     * The Ed25519 address.
+     */
+    readonly address: Ed25519Address;
+    /**
+     * @param address An Ed25519 address.
+     */
+    constructor(address: Ed25519Address) {
+        super(AddressType.ImplicitAccountCreation);
+        this.address = address;
+    }
+
+    toString(): string {
+        return this.address.toString();
+    }
+}
+
+/**
+ * An address with restricted capabilities.
+ */
+class RestrictedAddress extends Address {
+    /**
+     * The inner address.
+     */
+    readonly address: Address;
```

Comment:
this needs to be parsed again but will be a recursive issue. can look at the regulartransactionessence payload to see how we did it

Reply:
That was only a problem because the type was defined separately tho, you sure it's a problem here?

Reply:
@kwek20

Reply:
im referring to
```
@Type(() => Address, {
        discriminator: AddressDiscriminator,
    })
'''

Reply:
Since the inner address cannot be every address type anyway, I just created a new discriminator for it